### PR TITLE
feat(layer2): integrate moonriver and sushiswap dex

### PR DESCRIPTION
**Pull request by developerfred**
## Moonriver Network

Integration Moonriver Network

### Dex
 - [x] sushiswap
 - [x] solarbeam
 - [x] HuckleBerry

### Token List
- [ ] https://github.com/DimensionDev/Mask-Token-List/pull/50

### Contracts 
- [ ] [Balance Checker](https://github.com/wbobeirne/eth-balance-checker)  

### Mask Assets 
- [x] https://github.com/DimensionDev/assets/pull/6

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [x] I have removed all in development `console.log`s
  - [x] I have removed all commented code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

File: cspell.json [hunk 1]
```diff
@@ -12,53 +12,17 @@
         "*.snap"
     ],
     "words": [
-        "AMPL",
```
Comment (guanbinrui) [inline cspell.json]: Don't sort this.

File: packages/web3-constants/evm/airdrop.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
         "Arbitrum_Rinkeby": "",
         "xDai": "",
         "Celo": "",
-        "Fantom": ""
+        "Fantom": "",
+        "Moonriver": "1285"
```
Comment (guanbinrui) [inline packages/web3-constants/evm/airdrop.json]:
```suggestion
        "Moonriver": ""
```

File: packages/web3-constants/evm/ethereum.json [hunk 3]
```diff
@@ -13,7 +13,8 @@
         "Arbitrum_Rinkeby": "0xF98dC9391273c5ab9fa2c980A8d4F875ece802db",
         "xDai": "0x96c7D011cdFD467f551605f0f5Fce279F86F4186",
         "Celo": "0x8e28F1d64ceD52b9A09aB1AA3071Aa3c05802d1F",
-        "Fantom": "0xc119574d5fb333f5ac018658d4d8b5035e16bf39"
+        "Fantom": "0xc119574d5fb333f5ac018658d4d8b5035e16bf39",
+        "Moonriver": ""
```
Comment (guanbinrui) [inline packages/web3-constants/evm/ethereum.json:21]: @andy-at-mask @UncleBill

File: packages/web3-constants/evm/explorer.json [hunk 4]
```diff
@@ -29,6 +30,7 @@
         "Arbitrum_Rinkeby": "",
         "xDai": "",
         "Celo": "",
-        "Fantom": ""
+        "Fantom": "",
+        "Moonriver": ""
```
Comment (guanbinrui) [inline packages/web3-constants/evm/explorer.json]: `BMUVSDQX72NTGRC1KDNGD889NJKG6H85F2`

File: packages/web3-constants/evm/explorer.json [hunk 5]
```diff
@@ -13,7 +13,8 @@
         "Arbitrum_Rinkeby": "",
         "xDai": "",
         "Celo": "",
-        "Fantom": "https://api.ftmscan.com/api"
+        "Fantom": "https://api.ftmscan.com/api",
+        "Moonriver": ""
```
Comment (guanbinrui) [inline packages/web3-constants/evm/explorer.json]:
```suggestion
        "Moonriver": "https://api-moonriver.moonscan.io/api"
```

File: packages/web3-constants/evm/ethereum.json [hunk 6]
```diff
@@ -29,7 +30,8 @@
         "Arbitrum_Rinkeby": "0xa501c031958F579dB7676fF1CE78AD305794d579",
         "xDai": "0xAb7B1bE4233A04e5C43a810E75657ECED8E5463B",
         "Celo": "0x072453AdEC16cFC7FB6Af1517c3f25407180cccC",
-        "Fantom": "0x913975af2Bb8a6Be4100D7dc5e9765B77F6A5d6c"
+        "Fantom": "0x913975af2Bb8a6Be4100D7dc5e9765B77F6A5d6c",
+        "Moonriver": "0x43D002a2B468F048028Ea9C2D3eD4705a94e68Ae"
```
Comment (guanbinrui) [inline packages/web3-constants/evm/ethereum.json]: @UncleBill

File: packages/web3-constants/evm/token.json [hunk 7]
```diff
@@ -13,7 +13,8 @@
         "Arbitrum_Rinkeby": "0xB47e6A5f8b33b3F17603C83a0535A9dcD7E32681",
         "xDai": "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d",
         "Celo": "0x471EcE3750Da237f93B8E339c536989b8978a438",
-        "Fantom": "0x21be370d5312f44cb42ce377bc9b8a0cef1a4c83"
+        "Fantom": "0x21be370d5312f44cb42ce377bc9b8a0cef1a4c83",
+        "Moonriver": " "
```
Comment (guanbinrui) [inline packages/web3-constants/evm/token.json]: If the address is not provided, plz leave it empty string.

File: packages/web3-shared/evm/pipes/index.ts [hunk 8]
```diff
@@ -109,6 +111,7 @@ export const resolveChainColor = createLookupTableResolver<ChainId, string>(
         [ChainId.xDai]: 'rgb(73, 169, 166)',
         [ChainId.Celo]: 'rgb(53, 208, 127)',
         [ChainId.Fantom]: 'rgb(19, 181, 236)',
+        [ChainId.Moonriver]: 'rgb(242 183 6)',
```
Comment (guanbinrui) [inline packages/web3-shared/evm/pipes/index.ts]:
```suggestion
        [ChainId.Moonriver]: 'rgb(242, 183, 6)',
```

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Comment by guanbinrui** [comment]
I'm closing this PR. Cause it doesn't belong to any bounty issue.

**Pull request closed without merging**